includes/multi-currency/MultiCurrency.php

```diff
@@ -544,6 +548,40 @@ public function update_single_currency_settings( string $currency_code, string $
 		}
 	}
 
+	/**
+	 * Updates the customer currencies option.
+	 *
+	 * @param int $order_id The order ID.
+	 *
+	 * @return void
+	 */
+	public function maybe_update_customer_currencies_option( $order_id ) {
+		$order = wc_get_order( $order_id );
+
+		if ( ! $order ) {
+			return;
+		}
+
+		$currency   = $order->get_currency();
+		$currencies = self::get_all_customer_currencies();
+
+		// Skip if the currency is not a string or if the option is invalid.
+		if ( ! is_string( $currency ) || ! is_array( $currencies ) ) {
+			return;
+		}
+
+		$currency   = strtoupper( $currency );
+		$currencies = array_map( 'strtoupper', $currencies );
```

Comment:
It does make sense. I updated the code to transform the array before saving the option on https://github.com/Automattic/woocommerce-payments/pull/7143/commits/c947e6ce5bcd008aa3da77f2cdf3d322acac10af. Thanks for the suggestion!

Reply:
I thought about saving it directly as uppercase because the option wouldn't exist prior to this code execution, so it would always get the uppercase values. I do agree that it's better to guarantee that we're always dealing with uppercase currencies. The code was updated on https://github.com/Automattic/woocommerce-payments/pull/7143/commits/862dcbeacd61d2ef2e46e506e6241a3f725d36fe.